### PR TITLE
Initial plugin implementation

### PR DESCRIPTION
**Pull request by sabre1041**
Initial implementation of the `helm-sigstore` plugin and includes the following features:

* Uploading signed Helm Charts
* Searching for signed Helm Charts in Rekor
* Verifying signed Helm Charts

Includes documentation and necessary tooling to incorporate into the Helm plugin ecosystem. Current version uses a file path based reference to the plugin. Once a release is made, the plugin binary can be retrieved instead of requiring the user to build locally.

### CODE REVIEW

**Review by dlorenc — APPROVED** [review]

**Comment by dlorenc** [comment]
This looks great! I'm not really a helm expert but the rekor code looks right.

**Review by bobcallaway — APPROVED** [review]
LGTM

File: cmd/upload.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+)
+
+func NewUploadCmd() *cobra.Command {
+
+	uploadOptions := types.CLIOptions{}
+
+	// uploadCmd represents the upload command
+	uploadCmd := &cobra.Command{
+		Use:   "upload",
+		Short: "Upload Signed Helm Chart",
```
Comment (hectorj2f) [inline cmd/upload.go:38]: Would you like to add the `usage` ? I would expect to see something like `helm sigstore verify <path_to_chart>` like in cosign cli. Instead I see `Usage:  sigstore verify [flags]` which missed the required argument https://github.com/sigstore/helm-sigstore/pull/3/files#diff-6833c134d800a5e41f7be71a3764d426179a965ec90addbbc4fb724ae4890c64R42.

File: cmd/search.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+			if len(uuids) == 0 {
+				return errors.New(fmt.Sprintf("Unable to find an entry for Chart '%s'\n", chartManager.ChartPath))
+			}
+
+			fmt.Println("The Following Records were Found")
+
+			fmt.Println(fmt.Sprintf("\nRekor Server: %s", searchOptions.RekorServer))
+			for _, uuid := range uuids {
+				fmt.Println(fmt.Sprintf("Rekor UUID: %s", uuid))
```
Comment (hectorj2f) [inline cmd/search.go:70]: Wouldn't it be easier something like this `fmt.Printf("Rekor UUID: %s\n", uuid)` ?

**Review by hectorj2f — COMMENTED** [review]
@sabre1041 I tested and it worked for me. I just added a comment about the usage of the commands which is missing the required chart path.

File: cmd/upload.go (hunk 1 above)
Comment (sabre1041) [reply cmd/upload.go:38]: Updated

File: cmd/search.go (hunk 2 above)
Comment (sabre1041) [reply cmd/search.go]: Updated

**Review by hectorj2f — APPROVED** [review]
lgtm

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**